pallets/automation-time/src/lib.rs

```diff
@@ -562,14 +565,18 @@ pub mod pallet {
 		/// * Get the most recent timestamp from the block.
 		/// * Convert the ms unix timestamp to seconds.
 		/// * Bring the timestamp down to the last whole hour.
-		pub fn get_current_time_slot() -> Result<UnixTime, Error<T>> {
-			let now = <timestamp::Pallet<T>>::get().saturated_into::<UnixTime>();
+		pub fn get_current_time_slot() -> Result<UnixTime, DispatchError> {
```

Comment:
is this dispatch error correct? can't it also be Error<T> as well? does this not currently break compile for line 574?

Reply:
is this dispatch error correct?
Yes, it's OK!

can't it also be Error as well? 
NO, If it is Error here, the compilation will fail.
Error<T> does not contains ArithmeticError.

https://github.com/OAK-Foundation/OAK-blockchain/blob/6b5e7c9158028539079f959e4b3f9bae1b68d917/pallets/automation-time/src/lib.rs#L268

Refers to substrate code:
https://github.com/paritytech/substrate/blob/6a946fc36d68b89599d7ca1ab03803d10c78468c/frame/balances/src/lib.rs#L1001

does this not currently break compile for line 574?
No, It doesn't break compile.

Reply:
hm interesting. i guess different error types are fine then? have you spun it up locally and tried to trigger all of the errors (both the new Arithmetic error and the previous custom Error<T> errors)?

Reply:
The previous custom Error errors:

<img width="404" alt="image" src="https://user-images.githubusercontent.com/16951509/177910443-79a0538a-f480-4348-8a6c-5c8b5f6d1052.png">

Arithmetic error:

<img width="622" alt="image" src="https://user-images.githubusercontent.com/16951509/177911987-8e6eac5d-de89-4a81-91a6-da3b18e50b4d.png">